app/org/sagebionetworks/bridge/models/schedules/SchedulerFactory.java

```diff
@@ -0,0 +1,18 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import org.sagebionetworks.bridge.validators.ScheduleValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+public class SchedulerFactory {
+
+    public static TaskScheduler getScheduler(String schedulePlanGuid, Schedule schedule) {
+        // TODO: This validation will be moved to create/update, not the time of use.
+        Validate.entityThrowingException(new ScheduleValidator(), schedule);
+        
+        if (schedule.getCronTrigger() != null) {
+            return new CronTaskScheduler(schedulePlanGuid, schedule);
+        }
+        return new IntervalTaskScheduler(schedulePlanGuid, schedule);
```

Comment:
I like, a lot, in the schedulers, not having to pass the schedule from method to method. I would prefer to leave it that way... in truth, all the values could be passed in the constructor and getTasks() could take no arguments. But the version that allows some arguments was easiest to re-execute in tests. These scheduler objects won't be cached or re-used (creating an object like this is not expensive).